File: backend/src/main/java/com/fitmeai/service/UserService.java
package com.fitmeai.service;

import com.fitmeai.dto.request.UserUpdateRequest;
import com.fitmeai.dto.response.UserResponse;
import com.fitmeai.model.User;

public interface UserService {
    UserResponse getCurrentUserResponse(User user);
    UserResponse updateProfile(User user, UserUpdateRequest request);
}

File: frontEnd/src/app/core/services/user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';

export interface UserResponse {
  id: number;
  email: string;
  firstName: string;
  lastName: string;
  roles: string[];
}

export interface UserUpdateRequest {
  email: string;
  firstName: string;
  lastName: string;
}

@Injectable({
  providedIn: 'root'
})
export class UserService {
  private apiUrl = `${environment.apiUrl}/users`;

  constructor(private http: HttpClient) {}

  getProfile(): Observable<UserResponse> {
    return this.http.get<UserResponse>(`${this.apiUrl}/me`);
  }

  updateProfile(request: UserUpdateRequest): Observable<UserResponse> {
    return this.http.put<UserResponse>(`${this.apiUrl}/me`, request);
  }
}
